### ISSUE TITLE
Failed to get Kuberneted Ingress annotations

### ISSUE THREAD

**Issue by MonicaMagoniCom** [hashicorp/terraform-provider-kubernetes#2340]
Hi,

I have a Kubernetes Ingress but I need to retrieve its associated backend service. I try this:
data "kubernetes_ingress_v1" "ingress-v1" {
  metadata {
    name      = "usvc-ccs-ingress"
    namespace = "default"
    annotations = {
      "ingress.kubernetes.io/backends" = ""
    }
  }

data "google_compute_backend_service" "backend" {
  name       = element(keys(jsondecode(data.kubernetes_ingress_v1.ingress-v1.metadata.0.annotations["ingress.kubernetes.io/backends"])), 0)
  depends_on = [data.kubernetes_ingress_v1.ingress-v1]
}

But when I run terraform apply I get the error:

│ Error: Error in function call
│
│   on kubernetes/kube.data.tf line 13, in data "google_compute_backend_service" "backend":
│   13:   name = element(keys(jsondecode(data.kubernetes_ingress_v1.ingress-v1.metadata.0.annotations["ingress.kubernetes.io/backends"])), 0)
│     ├────────────────
│     │ while calling jsondecode(str)
│     │ data.kubernetes_ingress_v1.ingress-v1.metadata[0].annotations["ingress.kubernetes.io/backends"] is ""
│
│ Call to function "jsondecode" failed: EOF.

It seems to ignore the depends_on. How can I solve it?

**Comment by alexsomesan** [comment]
Thanks for raising this @MonicaMagoniCom. 
We are aware of similar issues concerning annotations in data sources and are looking at options to resolve it.

We will update here as soon as we have a fix.

**Comment by MonicaMagoniCom** [comment]
@alexsomesan also concerning NEGs: is there any way to retrieve NEG's name created from a Kubernetes service?

For example:
```
resource "kubernetes_service_v1" "usvc-nginx-internal" {
  metadata {
    name = "usvc-nginx-internal"
    annotations = {
      "cloud.google.com/neg" = "{\"ingress\": true}"
    }
  }
  spec {
    type             = "NodePort"
    port {
      port = 80
    }
    selector = {
      app : "usvc-nginx-app"
    }
  }
}
```

How can I retrieve the NEG's name? It seems to be a similar issue as the Ingress

**Comment by alexsomesan** [comment]
@MonicaMagoniCom just to be sure we're on the same page, what do you mean by NEG specifically?

**Comment by MonicaMagoniCom** [comment]
Yes. With the above snippet, I created a NEG from a Kubernetes service but I'm not able to reference it inside Terraform
